pkg/partition/partition.go

```diff
@@ -195,7 +195,54 @@ func Partition(sels []int64, diffs []bool, partitions []int64, vec *vector.Vecto
 			diffs[i] = diffs[i] || (v != w)
 			v = w
 		}
+	case types.T_decimal64:
```

Comment:
Should you rewrite the whole thing using generics?   The code looks exactly the same to me ...

Reply:
done